### PR TITLE
chore : Update CRC GitHub issue template to be more interactive (#4494)

### LINKED ISSUE #4494: [Doc] Improve github issue template to be more interactive
## Component

Project's GitHub Templates 

## Description

As discussed internally, the current issue template is using legacy workflow for issues templates.

We can update it to use [GitHub Forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) that have customizable web form fields.

We should update the GitHub issue template to use this web form style.

## Acceptance Criteria

- [ ] [Issue Template](https://github.com/crc-org/crc/blob/main/.github/ISSUE_TEMPLATE/bug_report.md) is updated to include web form format.

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Fixes: #4494

Relates to: #4494

CRC project is using quite old style issue template. Update it to be more interactive so that users can provide information easily.

You can check the preview of how it would look like [here](https://github.com/rohankanojia-forks/crc/issues/new/choose) .

I've only ported relevant fields that were present in old template. If you think something else needs to be added, please add a comment on the pull request.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change
- [X] Chore (non-breaking change which doesn't affect codebase;
  test, version modification, documentation, etc.)

## Proposed changes
Update GitHub Issue template to use new issue template format. 

## Testing
I've tested the template on my github fork. You can try it out [here](https://github.com/rohankanojia-forks/crc/issues/new/choose).

## Contribution Checklist
- [x] I have read the [contributing guidelines](https://github.com/crc-org/crc/blob/main/developing.adoc)
- [x] My code follows the style guidelines of this project
- [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
- [x] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I tested my code on specified platforms 
    - [ ] Linux
    - [ ] Windows
    - [ ] MacOS

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      description: A clear and concise description of what you expected to happen.
+      placeholder:
+    validations:
+      required: true
+
+  - type: textarea
+    id: actual-behavior
+    attributes:
+      label: Actual behavior
+      description: A clear and concise description of what actually happen.
```
Comment (vyasgun) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: nit: happens

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      label: CRC config
+      placeholder: |
+        # Put `crc config view` output here
+      description:
+      render: shell
+
+  - type: textarea
+    id: host-os
+    attributes:
+      label: Host Operating System
```
Comment (vyasgun) [inline .github/ISSUE_TEMPLATE/bug_report.yml:120]: Should this be a required field?

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+blank_issues_enabled: true
+contact_links:
+  - name: CRC IRC Community Chat (suitable for short, simple casual discussions)
+    url: https://freenode.net/
```
Comment (vyasgun) [inline .github/ISSUE_TEMPLATE/config.yml]: Is this the right URL?

Comment (rohanKanojia) [reply .github/ISSUE_TEMPLATE/config.yml]: I just used the same URL as provided in https://github.com/crc-org/crc?tab=readme-ov-file#community

I can change it to any other URL decided by the team.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (rohanKanojia) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Thanks for noticing, I copy pasted this template from [Eclipse JKube](https://github.com/eclipse-jkube/jkube).

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (vyasgun) [reply .github/ISSUE_TEMPLATE/config.yml]: No I mean the channel name is `#codeready` so there should be a clear direction to use that.

Comment (cfergeau) [reply .github/ISSUE_TEMPLATE/config.yml]: Didn't we move this to liberachat?
However nobody from the team is watching this channel as far as I know, so I'm not sure it is useful to mention it.

Comment (rohanKanojia) [reply .github/ISSUE_TEMPLATE/config.yml]: @cfergeau : Shall I remove this list element?

Comment (vyasgun) [reply .github/ISSUE_TEMPLATE/config.yml:4]: Just saw the `codeready` reference but I didn't notice it. Maybe we can increase the font or change its placement? Just my opinion

Comment (vyasgun) [reply .github/ISSUE_TEMPLATE/config.yml]: +1 with Rohan. I think we can remove this field

**Comment by vyasgun** [comment]
Looks good to me!

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (gbraad) [reply .github/ISSUE_TEMPLATE/config.yml]: We can always re-add after we made a decision in regard to #4506

**Review by gbraad — APPROVED** [review]

**Pull request merged**